resources/schema/Operations.md

```diff
@@ -0,0 +1,668 @@
+  }
+  ```
+
+## permissible_variables
+
+Returns the permissible variables for a given domain and standard
+
+- Input:
+
+  Target Domain: `LB`
+
+  Product: `sdtmig`
+
+  Version: `3-4`
+
+  ```yaml
+  - operation: permissible_variables`
+    id: $permissible_variables`
+  ```
+
+- Output:
+
+  ```json
+  ["LBGRPID", "LBREFID", "LBSPID", "..."]
+  ```
+
+## record_count
+
+Returns the number of records in the dataset
+
+## required_variables
+
+Returns the required variables for a given domain and standard
+
+- Input:
+
+  Target Domain: `LB`
+
+  Product: `sdtmig`
+
+  Version: `3-4`
+
+  ```yaml
+  - operation: required_variables`
+    id: $required_variables`
+  ```
+
+- Output:
+
+  ```json
+  ["STUDYID", "DOMAIN", "USUBJID", "LBSEQ", "LBTESTCD", "LBTEST"]
+  ```
+
+## study_domains
+
+Returns a list of the domains in the study
+
+## valid_codelist_dates
+
+Returns the valid codelist dates for a given standard
+
+Given a list of codelists:
+
+```json
+[
+  "sdtmct-2023-10-26",
+  "sdtmct-2023-12-13",
+  "adamct-2023-12-13",
+  "cdashct-2023-05-19"
+]
+```
+
+and standard: `sdtmig`
+
+the operation will return
+
+```json
+["2023-10-26", "2023-12-13"]
+```
+
+## valid_meddra_code_references
+
+Determines whether the values are valid in the following variables:
+
+- `--SOCCD`
+- `--HLGTCD`
+- `--HLTCD`
+- `--PTCD`
+- `--LLTCD`
+
+## valid_meddra_code_term_pairs
+
+Determines whether the values are valid in the following variable pairs:
+
+- `--SOCCD`, `--SOC`
+- `--HLGTCD`, `--HLGT`
+- `--HLTCD`, `--HLT`
+- `--PTCD`, `--DECOD`
+- `--LLTCD`, `--LLT`
+
+## valid_meddra_term_references
+
+Determines whether the values are valid in the following variables:
+
+- `--SOC`
+- `--HLGT`
+- `--HLT`
+- `--DECOD`
+- `--LLT`
+
+## valid_whodrug_references
+
+Checks if a reference to whodrug term in `name` points to the existing code in Atc Text (INA) file.
+
+## variable_count
+
+Returns a mapping of variable names to the number of times that variable appears in a domain within the study.
+
+- Input
+
+  ```json
+  {
+    "AE": ["STUDYID", "DOMAIN", "USUBJID", "AETERM", "AEENDTC"],
+    "LB": ["STUDYID", "DOMAIN", "USUBJID", "LBTESTCD", "LBENDTC"]
+  }
+  ```
+
+- Output
+
+  ```json
+  {
+    "STUDYID": 2,
+    "DOMAIN": 2,
+    "USUBJID": 2,
+    "--TERM": 1,
+    "--TESTCD": 1,
+    "--ENDTC": 2
+  }
+  ```
+
+## variable_exists
+
+Flag an error if MIDS is in the dataset currently being evaluated and the TM domain is not present in the study
+
+```yaml
+Rule Type: Domain Presence Check
+Check:
+  all:
+    - name: $MIDS_EXISTS
+      operator: equal_to
+      value: true
+    - name: TM
+      operator: not_exists
+Operations:
+  - id: $MIDS_EXISTS
+    name: MIDS
+    operator: variable_exists
+```
+
+## variable_is_null
+
+True if variable is missing or if all values within a variable are null or empty string
+
+## variable_names
+
+Return the set of variable names from the library for the given standard
+
+## variable_permissibilities
```

Comment:
This is now the VariableLibraryMetadata. It now requires a name specifying the value to get from the library metadata.

Reply:
Updated